pkg/controllers/node/termination/terminator/eviction.go

```diff
@@ -68,13 +69,15 @@ func IsNodeDrainError(err error) bool {
 
 type QueueKey struct {
 	types.NamespacedName
-	UID types.UID
+	UID      types.UID
+	nodeName string
```

Comment:
I think we should probably do providerID instead of node name. We already have the indexes on providerID for both nodeclaim and node, so it shouldn't be that much bigger of a deal.

Reply:
I agree, `providerID` would be better... How can we achieve this? The Pod resource does not have access to the providerID, except indirectly with the `nodeName`.

Is it safe to look up the node when we add a pod to the queue? Should the interface be modified to include the nodeClaim so we can take it off that resource? Any preferences?